codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpBoundProtocolGenerator.kt

```diff
@@ -926,7 +932,12 @@ private class ServerHttpBoundProtocolTraitImplGenerator(
                                 )
                             }
                         }
-                        rust("${symbolProvider.toMemberName(it.member)}.push(v);")
+                        val method = if (model.expectShape(it.member.target) is SetShape) {
+                            "insert"
```

Comment:
since we own `Set` why don't we add a push method for convenience? you can make it `#[doc(hidden)]` if you want